src/main/java/se/sundsvall/paratransit/integration/casedata/mapper/CaseDataMapper.java

```diff
@@ -27,43 +27,41 @@
+	public static PatchErrand toPatchErrand(final Errand errand, final String phase, final String displayPhase, final String phaseStatus, final String phaseAction) {
+		var patchErrand = toPatchErrand(errand, phase, phaseStatus, phaseAction);
+		return patchErrand.extraParameters(patchErrand.getExtraParameters().stream()
+			.filter(e -> !CASEDATA_KEY_DISPLAY_PHASE.equals(e.getKey()))
+			.collect(Collectors.toCollection(ArrayList::new)))
+			.addExtraParametersItem((new ExtraParameter(CASEDATA_KEY_DISPLAY_PHASE).values(displayPhase == null ? emptyList() : List.of(displayPhase))));
+	}
 
-		return patchErrand.extraParameters(parameters);
+	public static PatchErrand toPatchErrand(final Errand errand, final String phase, final String phaseStatus, final String phaseAction) {
+		return new PatchErrand()
+			.externalCaseId(errand.getExternalCaseId())
+			.phase(phase)
+			.facilities(null)
+			.extraParameters(ofNullable(errand.getExtraParameters()).orElse(emptyList()).stream()
+				.filter(e -> !CASEDATA_KEY_PHASE_STATUS.equals(e.getKey()))
+				.filter(e -> !CASEDATA_KEY_PHASE_ACTION.equals(e.getKey()))
+				.collect(Collectors.toCollection(ArrayList::new)))
+			.addExtraParametersItem(new ExtraParameter(CASEDATA_KEY_PHASE_STATUS).values(phaseStatus == null ? emptyList() : List.of(phaseStatus)))
+			.addExtraParametersItem(new ExtraParameter(CASEDATA_KEY_PHASE_ACTION).values(phaseAction == null ? emptyList() : List.of(phaseAction)));
 	}
 
-	public static PatchErrand toPatchErrand(final String externalCaseId, final String phase, final String phaseStatus, final String phaseAction, final List<ExtraParameter> extraParameters) {
-		final var patchErrand = new PatchErrand()
+	public static PatchErrand toPatchErrand(final Errand errand, final String externalCaseId, final String phaseAction) {
+		return new PatchErrand()
 			.externalCaseId(externalCaseId)
-			.phase(phase);
-
-		var parameters = Optional.ofNullable(extraParameters).orElse(emptyList()).stream()
-			.filter(extraParameter -> !CASEDATA_KEY_PHASE_STATUS.equals(extraParameter.getKey()) && !CASEDATA_KEY_PHASE_ACTION.equals(extraParameter.getKey()))
-			.collect(Collectors.toCollection(ArrayList::new));
-
-		Optional.ofNullable(phaseStatus).ifPresentOrElse(status -> parameters.add(new ExtraParameter(CASEDATA_KEY_PHASE_STATUS).values(List.of(status))),
-			() -> parameters.add(new ExtraParameter(CASEDATA_KEY_PHASE_STATUS).values(emptyList())));
-
-		// Cannot be null since that could erase action when it is "AUTOMATIC"
-		Optional.ofNullable(phaseAction).ifPresentOrElse(action -> parameters.add(new ExtraParameter(CASEDATA_KEY_PHASE_ACTION).values(List.of(action))),
-			() -> { throw new IllegalArgumentException("phaseAction cannot be null"); });
-
-		return patchErrand.extraParameters(parameters);
+			.facilities(null)
+			.extraParameters(ofNullable(errand.getExtraParameters()).orElse(emptyList()).stream()
+				.filter(e -> !CASEDATA_KEY_PHASE_ACTION.equals(e.getKey()))
```

Comment:
Just something that may enhance readability (according to my personal opinion):

`.filter(e -> !CASEDATA_KEY_PHASE_ACTION.equals(e.getKey()))`
->
`.filter(not(e -> CASEDATA_KEY_PHASE_ACTION.equals(e.getKey()))) // Import from java.util.function.Predicate`  

or maybe:

```
.filter(not(this::isPhaseActionKey))

private boolean isPhaseActionKey(Map.Entry<String, ?> entry) {
    return CASEDATA_KEY_PHASE_ACTION.equals(entry.getKey());
} 
```

Do as you like with this comment...